src/pylibftdi/device.py

```diff
@@ -128,6 +128,8 @@ def __init__(
         encoding: str = "latin1",
         interface_select: int | None = None,
         device_index: int = 0,
+        vid: int | None = None,
+        pid: int | None = None,
```

Comment:
Having lots of positional arguments isn't great as they make maintaining backwards compatibility harder in the event someone uses them as positional args.

Could you either precede these with a '*' to make them kwarg-only (probably the right long term thing is to move all the other args from `kwargs` into explicit kwarg-only arguments) or handle in the same way as the other args (e.g. `index` => self.list_index on line 213) as the smaller change for now.

Either way, the new arguments need adding at the end of the docstring (and ideally in the docs, but I'm happy to update those as part of a release).

Also is it likely that users will want to specify only one of VID / PID? If they will (almost) always be specified together, having a single vid_pid pair is probably better IMO.

Reply:
I made them part of kwargs similar to `index` and added docstrings. 

As for whether they should be specified together or separate, I think it would be good to keep them separate to enable flexibility.

For example, in our specific use case the vendor ID is already supported in the existing library, but not the product ID. 

While it wouldn't be very difficult for users to determine both for their specific device, I think it's ultimately more convenient and user-friendly to allow them to be passed in independently. However, if you envision this as potentially generating instability, I would be open to the other implementation.